include/envoy/buffer/buffer.h

```diff
@@ -168,6 +168,9 @@ class Instance {
    */
   virtual Api::SysCallIntResult read(int fd, uint64_t max_length) PURE;
 
+  virtual Api::SysCallIntResult recvFrom(int fd, uint64_t max_length,
```

Comment:
nit: doc comments. As a larger comment, I would love to remove any type of file IO from the buffer interface moving forward since in the long term I don't think it fits with our FD refactors. Any thoughts on how we might not add it for this case since it's new? cc @brian-pane who is working on a new buffer impl.